### ISSUE TITLE
Panic on (maybe) moving a bunch of files/dirs from content directory 

### ISSUE THREAD

**Issue by focusaurus** [getzola/zola#559]
# Bug Report

## Environment

Zola version: 0.5.0

## Expected Behavior
`zola serve` gracefully handles changes to the content filesystem and continues watching and rebuilding.

## Current Behavior
```-> Content changed /host/content/problog/_index.md
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', libcore/option.rs:345:21
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::panicking::default_hook::{{closure}}
             at libstd/sys_common/backtrace.rs:71
   2: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:227
             at libstd/panicking.rs:477
   3: std::panicking::continue_panic_fmt
             at libstd/panicking.rs:391
   4: rust_begin_unwind
             at libstd/panicking.rs:326
   5: core::panicking::panic_fmt
             at libcore/panicking.rs:77
   6: core::panicking::panic
             at libcore/panicking.rs:52
   7: library::content::ser::SerializingSection::from_section
   8: site::Site::register_tera_global_fns
   9: zola::cmd::serve::serve
  10: zola::main
  11: std::rt::lang_start::{{closure}}
  12: main
  13: __libc_start_main
  14: <unknown>
```
## Step to reproduce

Not totally sure, but my guess is this was caused when I `mv`ed a bunch of directories containing content `.md` file from the `content` directory out to `/tmp` so I could regenerate them.

**Comment by Keats** [comment]
Can you try with 0.5.1? I fixed some bugs around that.

**Comment by jonas-schievink** [comment]
I saw something that looks very similar on 0.5.1: 

```
-> Content changed /home/jonas/dev/blog-zola/content/post/_index.md
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', libcore/option.rs:355:21
```

Unfortunately I didn't capture a backtrace.

**Comment by Keats** [comment]
I'm guessing it would be the same backtrace.
I think the rebuild fn in some cases forgets to update the sections that some pages has changed version while the rest of the code expects it to be always up to date (hence the unwrap). This is a bit weird as it should reload in all cases but some cases must have slipped through.

**Comment by focusaurus** [comment]
I don't have steps to reproduce unfortunately as I was just working away on my project not anticipating zola crashes. I've been on 0.5.1 for the most recent batch of work though and haven't seen this issue again.

**Comment by Keats** [comment]
Should be fixed with 0.6.0
